buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -1392,5 +1395,25 @@ private static boolean isUtf8(ByteBuf buf, int index, int length) {
         return true;
     }
 
+    /**
+     * Read bytes from the given {@link ByteBuffer} into the given {@link OutputStream} using the {@code position} and
+     * {@code length}. The position and limit of the given {@link ByteBuffer} may be adjusted.
+     */
+    static void readBytes(ByteBuffer buffer, int position, int length, OutputStream out) throws IOException {
+        if (buffer.hasArray()) {
+            out.write(buffer.array(), position + buffer.arrayOffset(), length);
+        } else {
+            byte[] tmp = new byte[Math.min(length, WRITE_CHUNK_SIZE)];
```

Comment:
so you decided against using alloc() to get this temp array?

Reply:
let me check if its possible here as well (in the `Unsafe` portion it is.

Reply:
fixed @CodingFabian